src/features/categories/category-selection/CategorySelect.tsx

```diff
@@ -65,7 +69,7 @@ const CategorySelect = ({
                     <div className={classes.flexContainerItem} key={cat.id.toString()}>
                       <CategoryBadge
                         {...cat}
-                        onClick={() => handleCategorySelection(cat.id, 'remove')}
+                        onClick={!readOnly ? () => handleCategorySelection(cat.id, 'remove') : undefined}
```

Comment:
This could be more elegant.
Replace this line with: `onClick={() => !readOnly && handleCategorySelection(cat.id, 'remove')}`